### DISCUSSION TITLE
Running YOLOv5 on Jetson Nano

### DISCUSSION THREAD

**Discussion by HuumbleBee** [ultralytics/yolov5#7348] [Q&A] [answered]
Sorry, i need help. i have problem when i want to running Yolov5 on my Jetson nano device. it show an error when i running it. can anyone want to help me what should i do to fix it? thanks

/home/jetson/.local/lib/python3.9/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: 
  warn(f"Failed to load image Python extension: {e}")
detect: weights=22Feb.pt, source=/home/jetson/Desktop/yolov5-master/data/images/zidane.jpg, imgsz=[640, 640], conf_thres=0.25, iou_thres=0.45, max_det=1000, device=, view_img=False, save_txt=False, save_conf=False, save_crop=False, nosave=False, classes=None, agnostic_nms=False, augment=False, visualize=False, update=False, project=runs/detect, name=exp, exist_ok=False, line_thickness=3, hide_labels=False, hide_conf=False, half=False, dnn=False
YOLOv5 🚀 2021-12-22 torch 1.11.0 CPU

Fusing layers... 
Model Summary: 232 layers, 7257306 parameters, 0 gradients
Traceback (most recent call last):
  File "/home/jetson/Desktop/yolov5-master/detect.py", line 307, in <module>
    main(opt)
  File "/home/jetson/Desktop/yolov5-master/detect.py", line 302, in main
    run(**vars(opt))
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/jetson/Desktop/yolov5-master/detect.py", line 122, in run
    pred = model(im, augment=augment, visualize=visualize)
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/jetson/Desktop/yolov5-master/models/common.py", line 384, in forward
    y = self.model(im) if self.jit else self.model(im, augment=augment, visualize=visualize)
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/jetson/Desktop/yolov5-master/models/yolo.py", line 126, in forward
    return self._forward_once(x, profile, visualize)  # single-scale inference, train
  File "/home/jetson/Desktop/yolov5-master/models/yolo.py", line 149, in _forward_once
    x = m(x)  # run
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/nn/modules/upsampling.py", line 154, in forward
    recompute_scale_factor=self.recompute_scale_factor)
  File "/home/jetson/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1185, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'Upsample' object has no attribute 'recompute_scale_factor'

**Comment by glenn-jocher** [comment, marked as answer]
@HuumbleBee good news 😃! Your original issue is fixed ✅ in PR #6932 about a month ago. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Reply by HuumbleBee** [reply]
thanks

**Comment by zeleny-ondrej** [comment]
Did you managed to run it on Jetson Nano ? What is your setup ? I have a 2GB Jetson Nano and for some reason the model Ensemble() takes up so much space it gets killed.

**Reply by glenn-jocher** [reply]
@zeleny-ondrej ensembling models is probably not something you should be doing if you are trying to run on low-resource devices.

**Reply by zeleny-ondrej** [reply]
I tried to do it the same way as its done in training loop in case of resuming, however that did help just slightly. I tried looking up how much space is allocated when i was running it on PC and it was about 700MB of GPU memory. I checked on Jetson Nano and i had 1.5GB and it didnt fit. I runs on CPU just fine but the inference time is about 1 frame/sec.
